WordPress/Classes/ViewRelated/Blog/Blogging Reminders/BloggingRemindersFlowSettingsViewController.swift

```diff
@@ -227,31 +252,67 @@ class BloggingRemindersFlowSettingsViewController: UIViewController {
 
-        let topRow = days[0..<4]    // First 4 days
-        let bottomRow = days[4..<days.count]    // Last 3 days
-        daysTopInnerStackView.addArrangedSubviews(topRow.map({ CalendarDayToggleButton(weekday: $0) }))
-        daysBottomInnerStackView.addArrangedSubviews(bottomRow.map({ CalendarDayToggleButton(weekday: $0) }))
-    }
-
-    /// Localized short weekday names, starting at the correct weekday for the current calendar
-    private let days: [String] = {
-        var calendar = Calendar.current
-        calendar.locale = Locale.autoupdatingCurrent
-        let firstWeekday = calendar.firstWeekday
-        var symbols = calendar.shortWeekdaySymbols
+        let createButton: (Int) -> CalendarDayToggleButton = { [unowned self] calendarWeekday in
+            let isSelected: Bool
+            let rawValue = (calendarWeekday + calendar.firstWeekday) % calendar.shortWeekdaySymbols.count
+
+            let weekday = BloggingRemindersScheduler.Weekday(rawValue: rawValue) ?? {
+                // The rawValue above should always fall within 0 ..< 7 and this case
+                // should not be possible.  We're still going to handle it by logging
+                // an error and returning .monday so that the app keeps running.
+                //
+                // Is there any other sensible approach to handle this?
+                //
+                DDLogError("Cannot initialize BloggingRemindersScheduler.Weekday with rawValue: \(rawValue)")
+                return .monday
+            }()
+
+            isSelected = weekdays.contains(weekday)
+
+            return CalendarDayToggleButton(
+                weekday: weekday,
+                dayName: self.calendar.shortWeekdaySymbols[calendarWeekday],
+                isSelected: isSelected) { button in
+
+                if button.isSelected {
+                    self.weekdays.append(button.weekday)
+                } else {
+                    self.weekdays.removeAll { weekday in
+                        weekday == button.weekday
+                    }
+                }
+            }
+        }
 
-        // Switch around the order of days so that the correct day is at the beginning
-        let firstWeekdayToEnd = symbols[firstWeekday-1 ..< Calendar.current.shortWeekdaySymbols.count]
-        let beginningToFirstWeekday = symbols[0 ..< firstWeekday - 1]
+        let topRow = 0 ..< 4
+        let bottomRow = 4 ..< calendar.shortWeekdaySymbols.count
 
-        return Array(firstWeekdayToEnd + beginningToFirstWeekday)
-    }()
+        daysTopInnerStackView.addArrangedSubviews(topRow.map({ createButton($0) }))
+        daysBottomInnerStackView.addArrangedSubviews(bottomRow.map({ createButton($0) }))
+    }
 
     // MARK: - Actions
 
     @objc private func notifyMeButtonTapped() {
         tracker.buttonPressed(button: .continue, screen: .dayPicker)
 
+        do {
+            try scheduler.schedule(.weekdays(weekdays))
```

Comment:
This is where the scheduling should happen.  The issue is that I designed this method without a completion block.

My thinking is that I need to add a completion block here, and handle any errors in the notifications scheduling, but I'm not sure what we should do if, for example, a single notification raises an error when we try to schedule it.

Do we show the user an error?  Do we silently ignore the error?

Reply:
We decided to leave this as-is for now, and review the error handling at another time.  This should be fine as the feature is not public.